backend/tarsy/services/alert_service.py

```
@@ -363,64 +354,58 @@ async def _execute_chain_stages(
         Extract final analysis from stages for API consumption.
         
         Uses the final_analysis field which contains clean, concise summaries
         extracted by each agent's iteration controller.
         """
         # Look for final_analysis from the last successful stage (typically a final-analysis stage)
-        for stage_name in reversed(list(alert_data.stage_outputs.keys())):
-            stage_result = alert_data.stage_outputs[stage_name]
+        for stage_name in reversed(list(chain_context.stage_outputs.keys())):
+            stage_result = chain_context.stage_outputs[stage_name]
             if isinstance(stage_result, AgentExecutionResult):
                 if stage_result.status.value == "completed" and stage_result.final_analysis:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)